lib/computer.rb

```diff
@@ -1,15 +1,50 @@
-require "./spec/spec_helper"
-
 class Computer
-  attr_reader :mark
+  attr_reader :mark, :opponent
 
-  def initialize(mark)
-    @moves = computer_player
+  def initialize(mark, board)
     @mark = mark
+    @board = board
   end
 
   def get_move
-    @moves.shift
-    # return a random available move
+    analyse_board
+  end
+
+  def opponent_mark
+    if @mark == "O"
+      opponent = "X"
+    else 
+      opponent = "O"
+    end
+    opponent
+  end
+
+  def analyse_board
+    spaces = @board.available_moves
+    current_board = @board.board_array
+    best_play = nil 
+
+    spaces.each do |space|
+      @board.mark_board(space, @mark)
+      if @board.current_player_wins?(@mark)
+        best_play = space
+        @board.board_array[space.to_i - 1] = space
+      else
+        @board.board_array[space.to_i - 1] = space
+      end
+      @board.mark_board(space, opponent_mark)
+      if @board.current_player_wins?(opponent_mark)
+        best_play = space
+        @board.board_array[space.to_i - 1] = space
+      else
+        @board.board_array[space.to_i - 1] = space
+      end 
+     end
+   
+    if best_play 
+      return best_play
```

Comment:
Why the return here? If you had it on  47 for it to be symmetrical I probably wouldn't call it out, but this jumps out to me. I'd personally remove the return instead of adding one below. There's a reason for that which I'd like you to research and explain to me.

Reply:
Resolved #32